### PR TITLE
Build with cmake

### PR DESCRIPTION
**Pull request by iamashwin99**
Depends on https://github.com/fangohr/octopus-in-docker/pull/31

Cmake build commands thanks to @LecrisUT

### CODE REVIEW

**Comment by iamashwin99** [comment]
@fangohr any thoughts on this mr?
We can merge this so that we can create a branch `cmake_main` from `main` which would then be used to carry out further CI runs to test the cmake-related problems.

In the mean time, development on main can carry on as the cmake pipelines are set to only trigger on a branch named `cmake_main`

**Comment by fangohr** [comment]
Yes, great. Is this ready for review?

**Comment by iamashwin99** [comment]
yes

File: README.rst [hunk 1]
```diff
@@ -165,6 +165,14 @@ or use ``uname -m`` inside the container.)
 .. |develop| image:: https://github.com/fangohr/octopus-in-docker/actions/workflows/develop.yml/badge.svg
    :target: https://github.com/fangohr/octopus-in-docker/actions/workflows/debian-develop.yml
 
+A note on Build System
+-----------------------
+Octopus from 14.0 onwards supports CMake as a build system. The Dockerfile uses the build arg `BUILD_SYSTEM` to specify the build system.
+The default value is `autotools`. If you want to use CMake as the build system pass `--build-arg BUILD_SYSTEM=cmake` to the `docker build` command.
```
Comment (fangohr) [inline README.rst:171]:
```suggestion
The default value is ``autotools``. If you want to use CMake as the build system, then pass ``--build-arg BUILD_SYSTEM=cmake`` to the ``docker build`` command.
```

File: README.rst [hunk 2]
```diff
@@ -165,6 +165,14 @@ or use ``uname -m`` inside the container.)
...
+Alternatively, you can set the environment variable `BUILD_SYSTEM` to `cmake`. For eg:
```
Comment (fangohr) [inline README.rst:172]:
```suggestion
Alternatively, you can set the environment variable ``BUILD_SYSTEM`` to ``cmake``. For eg:
```
Please check the rendered output (https://github.com/fangohr/octopus-in-docker/tree/build-with-cmake). I hope my suggested changes fixes it.

File: README.rst [hunk 3]
```diff
@@ -165,6 +165,14 @@ or use ``uname -m`` inside the container.)
 .. |develop| image:: https://github.com/fangohr/octopus-in-docker/actions/workflows/develop.yml/badge.svg
    :target: https://github.com/fangohr/octopus-in-docker/actions/workflows/debian-develop.yml
 
+A note on Build System
```
Comment (fangohr) [inline README.rst:168]:
```suggestion
CMAKE or autotools as the configuration and build system
```

File: README.rst [hunk 4]
```diff
@@ -165,6 +165,14 @@ or use ``uname -m`` inside the container.)
...
+-----------------------
+Octopus from 14.0 onwards supports CMake as a build system. The Dockerfile uses the build arg `BUILD_SYSTEM` to specify the build system.
```
Comment (fangohr) [inline README.rst:170]:
```suggestion
Octopus from 14.0 onwards supports CMake as a build system. The Dockerfile uses the build arg ``BUILD_SYSTEM`` to specify the build system.
```

File: README.rst [hunk 5]
```diff
@@ -165,6 +165,14 @@ or use ``uname -m`` inside the container.)
 .. |develop| image:: https://github.com/fangohr/octopus-in-docker/actions/workflows/develop.yml/badge.svg
    :target: https://github.com/fangohr/octopus-in-docker/actions/workflows/debian-develop.yml
 
+A note on Build System
+-----------------------
```
Comment (fangohr) [inline README.rst:169]:
```suggestion
--------------------------------------------------------
```

**Review by fangohr — APPROVED** [review]
Nice!

Some fixes, mostly relating to this being rst (not markdown), we need two backticks to enclose monospace statements.

Finally, I guess we should have status badges for the CMAKE builds? We could add those later when the CMake works.

I approve so you can merge after addressing the suggestions.

**Comment by iamashwin99** [comment]
Sorry, I mixed up the rst and md syntax.

**Pull request merged**